File: src/main/java/com/arnas/klatrebackend/interfaces/services/PlaceService.java
package com.arnas.klatrebackend.interfaces.services;

import com.arnas.klatrebackend.dataclasses.Place;
import com.arnas.klatrebackend.dataclasses.PlaceWithGrades;
import com.arnas.klatrebackend.records.PlaceDTO;

import java.util.List;

public interface PlaceService {
    Place getPlaceById(long placeId);
    List<PlaceWithGrades> getPlacesByGroupId(long groupId, long userId);
    int updatePlace(long userId, PlaceDTO placeUpdateDTO);
    int updatePlaceGradingSystem(long userId, long placeId, long newGradingSystemId);
}

File: src/main/kotlin/com/arnas/klatrebackend/features/places/PlaceService.kt
package com.arnas.klatrebackend.features.places

interface PlaceService {
    fun getPlaceById(placeId: Long): Place?
    fun getPlacesByGroupId(groupId: Long, userId: Long): List<PlaceWithGrades>
    fun updatePlace(userId: Long, placeUpdateDTO: PlaceUpdateDTO)
    fun updatePlaceGradingSystem(placeId: Long, oldGradingSystemId: Long, newGradingSystemId: Long?): Long

}